src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogPostForMailItem.java

```diff
@@ -1,29 +1,111 @@
+ */
+@Getter
 class BlogPostForMailItem {
-
+    
+    private static final String UTM_SOURCE = "jvmbloggers";
+    private static final String UTM_MEDIUM = "newsletter";
+    
     private String title;
     private String url;
     private String authorLabel;
 
-    public BlogPostForMailItem(BlogPost blogPost) {
-        this.title = blogPost.getTitle();
-        this.url = blogPost.getUrl();
-        this.authorLabel = determineAuthorLabel(blogPost.getBlog());
+    /**
+     * Returns internal builder.
+     * 
+     * @return builder
+     */
+    public static Builder builder() {
+        return new Builder();
     }
+    
+    /**
+     * Builder for {@link BlogPostForMailItem}
+     */
+    public static class Builder {
 
-    private String determineAuthorLabel(Blog blogger) {
-        if (blogger.getTwitter() != null) {
-            return "<a href=\"https://twitter.com/" + blogger.getTwitter().substring(1) + "\">" + blogger.getAuthor() + "</a>";
-        } else {
-            return blogger.getAuthor();
+        private final BlogPostForMailItem instance = new BlogPostForMailItem();
+
+        public Builder from(BlogPost blogPost) {
+            instance.title = blogPost.getTitle();
+            instance.url = blogPost.getUrl();
+            instance.authorLabel = determineAuthorLabel(blogPost.getBlog());
+            return this;
+        }
+        
+        public Builder withTitle(String title) {
+            instance.title = title;
+            return this;
         }
-    }
 
+        public Builder withUrl(String url) {
+            instance.url = url;
+            return this;
+        }
+        
+        public Builder withAuthorLabel(Blog blog) {
+            instance.authorLabel = determineAuthorLabel(blog);
+            return this;
+        }
+        
+        public Builder withUrlParameter(String name, String value) {
+            Preconditions.checkState(
+                instance.url != null,
+                "Url could not be null. Please set url first"
+            );
+            
+            instance.url = UriComponentsBuilder
+                .fromHttpUrl(instance.url)
+                .queryParam(name, value)
+                .build().toString();
+            
+            return this;
+        }
+        
+        public Builder withDefaultUTMParameters() {
+            withUrlParameter(UTMParametrsConstants.UTM_SOURCE_KEY, UTM_SOURCE);
+            withUrlParameter(UTMParametrsConstants.UTM_MEDIUM_KEY, UTM_MEDIUM);
+            withUrlParameter(
+                UTMParametrsConstants.UTM_CAMPAING_KEY,
+                String.format(
+                    "%s#%s", UTM_SOURCE,
+                    new SimpleDateFormat("yyyy-mm-dd").format(new Date())
+                )
+            );
+            
+            return this;
+        }
+        
+        public BlogPostForMailItem build() {
+            Preconditions.checkState(instance.title != null, "Tittle cannot be null");
```

Comment:
If you remove all unused methods from my comment above, this check could be moved to from() method. But I am not sure it that was your original idea, so it is just a suggestion.